com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/util/ReadOnlyParametersChecker.java

```diff
@@ -0,0 +1,65 @@
+package com.sap.cloud.lm.sl.cf.process.util;
+
+import java.text.MessageFormat;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.inject.Named;
+
+import com.google.common.collect.Sets;
+import com.sap.cloud.lm.sl.cf.core.model.SupportedParameters;
+import com.sap.cloud.lm.sl.cf.process.message.Messages;
+import com.sap.cloud.lm.sl.common.SLException;
+import com.sap.cloud.lm.sl.mta.model.DeploymentDescriptor;
+import com.sap.cloud.lm.sl.mta.model.ParametersContainer;
+
+@Named
+public class ReadOnlyParametersChecker {
```

Comment:
Let's say we have 2 modules and 2 resources, all with invalid parameters. With the current logic, the mta deploy will fail 4 times and every time the customer should adjust something new.
It would be better if they receives faster feedback.

Other idea for improvement is that parameter is defined which module/resource has define it. If you like that idea, it will exclude the other one, I guess. Because we have to know currently processed module or resource.